src/mats_utils/geolocation/coordinates.py

```diff
@@ -178,4 +181,204 @@ def angles(ccditem):
+    y_step : int
+        step along the y-axis in the image between 2 sampled points used for interpolation. The default value is 2.
+            
+    Returns
+    -------
+    lat_map : array[float]
+        map giving the latitude for each pixel in the image
+    lon_map : array[float]
+        map giving the longitude for each pixel in the image
+    sza_map : array[float]
+        map giving the solar zenith angle for each pixel in the image
+    """
+    im = ccditem['IMAGE']
+    x_deg_map, y_deg_map = deg_map(ccditem) # creating angle deviation map for each pixel (degress)
+    a,b = np.shape(im)
+
+    metoOHB  = R.from_matrix([[0,0,-1],[0,-1,0],[-1,0,0]])
+    ts=sfapi.load.timescale()
+    t=ts.from_datetime(ccditem['EXPDate'].replace(tzinfo=sfapi.utc)) # exposure time  
+    q=ccditem.afsAttitudeState
+    quat=R.from_quat(np.roll(q,-1)) # quaternion of MATS attitude (for the LIMB imager) 
+    pos=ccditem.afsGnssStateJ2000[0:3] # position of MATS
+    
+    xd = range(0,b,x_step) # sampled pixels on the x axis
+    yd = range(0,a,y_step) # sampled pixels on the y axis
+    LAT = np.zeros((len(yd),len(xd)))
+    LON = np.zeros((len(yd),len(xd)))
+    SZA = np.zeros((len(yd),len(xd)))
+
+    # computing the latitude, longitude and solar zenith angles at the intersection of the line of sight and the earth surface
+    # only the line of sights from some sampled pixels are computed
+    for i in range(len(yd)):
+        for j in range(len(xd)):
+                x = xd[j]
+                y = yd[i]
+                # angular transformations
+                # rotation from the line of sight of the LIMB imager to the line of sight of the NADIR pixel
+                angle = R.from_euler('XYZ', [x_deg_map[y,x],-(90-23)+y_deg_map[y,x],0] , degrees=True).apply([1, 0, 0])
+                FOV = quat.apply(metoOHB.apply(angle)) # attitude state for the line of sight of the NADIR pixel    
+                # finding the distance between the point pos and the Geoid along the line of sight
+                res = findsurface(t,pos,FOV)
+                newp = pos + res.x * FOV 
+                newp = ICRF(Distance(m=newp).au, t=t, center=399) # point at the intersection between the line of sight at the pixel and the Geoid surface
+                LAT[i,j]=wgs84.subpoint(newp).latitude.degrees # latitude of the point
+                LON[i,j]=wgs84.subpoint(newp).longitude.degrees # longitude of the point    
+
+                # finding the solar zenith angle of the point
+                planets = sfapi.load('de421.bsp')
+                earth=planets['Earth']
+                sun=planets['Sun']
+                SZA[i,j]=90-((earth+wgs84.subpoint(newp)).at(t).observe(sun).apparent().altaz())[0].degrees
+    
+    # interpolating the results along all the pixels
+    interp_lat = RegularGridInterpolator((yd,xd),LAT,method="quintic",bounds_error=False,fill_value=None) 
```

Comment:
not used this  RegularGridInterpolator , I am sure it is fine -- Could you use bigger steps above?

Reply:
With the 'quintic' interpolation at least 6 points on each axis are needed, I will add the interpolation method as an optional parameter. It is possible to use cubic and linear interpolation

Reply:
up to you